### PR TITLE
Add CLI and cmake option for hipify-torch

### PR DESCRIPTION
**Pull request by jithunnair-amd**
Added Hipify.cmake and hipify_cli.py

Credit: Michael Melesse (@micmelesse)

### CODE REVIEW

**Comment by pruthvistony** [comment]
The code changes are fine.

One comment I have, is regarding the hipify submodule clone directory, 

For Eg: consider 
/tensorpipe/third_party/hipify-torch/hipify_cli.py
/tensorpipe/third_party/hipify-torch/cmake/Hipify.cmake
/tensorpipe/third_party/hipify-torch/hipify/hipify_python.py

For above directory structure it will break, since Hipify.cmake will not able to find hipify_cli.py.

File: hipify_cli.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+# LICENSE file in the root directory of this source tree.
+
+import os
+import sys
+import argparse
+from hipify import hipify_python
+
+parser = argparse.ArgumentParser(
+    description='Top-level script for HIPifying, filling in most common parameters')
+parser.add_argument(
```
Comment (pruthvistony) [inline hipify_cli.py:16]: I have used similar structure and parameter passing in tensorpipe and it is working. 
LGTM.
Only thing I havent used is --ignores and --includes.

**Review by pruthvistony — COMMENTED** [review]
Please check directory structure when Hipify.cmake will not be able to find hipify_cli.py.

**Comment by jithunnair-amd** [comment]
> Please check directory structure when Hipify.cmake will not be able to find hipify_cli.py.

Would it not be feasible to use one of the inbuilt CMake variables (https://gitlab.kitware.com/cmake/community/-/wikis/doc/cmake/Useful-Variables) to be able to locate the hipify_cli.py consistently?

Ideally, we'd want something that would take either the directory where the Hipify.cmake resides and do a relative path to get to the hipify_cli.py, or start from the hipify-torch root directory and work downwards.

**Comment by jithunnair-amd** [comment]
@pruthvistony Can you add the cmake utility functions to this same PR branch, or did you want that as a separate PR?

**Comment by pruthvistony** [comment]
> > Please check directory structure when Hipify.cmake will not be able to find hipify_cli.py.
> 
> Would it not be feasible to use one of the inbuilt CMake variables (https://gitlab.kitware.com/cmake/community/-/wikis/doc/cmake/Useful-Variables) to be able to locate the hipify_cli.py consistently?
> 
> Ideally, we'd want something that would take either the directory where the Hipify.cmake resides and do a relative path to get to the hipify_cli.py, or start from the hipify-torch root directory and work downwards.

Need to check if CMAKE_CURRENT_SOURCE_DIR works, AFAIK *.cmake file is included into the main file, so not sure of this value. Will experiment on this and update this PR.

**Comment by pruthvistony** [comment]
> @pruthvistony Can you add the cmake utility functions to this same PR branch, or did you want that as a separate PR?

I feel it is better to keep cmake utility functions as a separate PR.

File: cmake/Hipify.cmake [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # cmake file to trigger hipify
 
-set(HIPIFY_SCRIPTS_DIR ${PROJECT_SOURCE_DIR})
+# Get the hipify_cli.py source directory from current directory
+get_filename_component(HIPIFY_SCRIPTS_DIR ${CMAKE_CURRENT_LIST_DIR} DIRECTORY)
```
Comment (jithunnair-amd) [inline cmake/Hipify.cmake:4]: CMake noob question: wouldn't `set(HIPIFY_SCRIPTS_DIR ${CMAKE_CURRENT_LIST_DIR}` work?

Comment (pruthvistony) [reply cmake/Hipify.cmake:4]: Consider hipify-torch is clone at 
/tensorpipe/third_party/hipify-torch
then 
`CMAKE_CURRENT_LIST_DIR = /tensorpipe/third_party/hipify-torch/cmake/Hipify.cmake` (in hipify.cmake file)
to get to parent dir, to access hipify_cli.py, we need the `get_filename_component()` which will set `HIPIFY_SCRIPTS_DIR = /tensorpipe/third_party/hipify-torch/`

**Pull request merged**